### PR TITLE
#22 enable join button earlier

### PR DESCRIPTION
**Pull request by aaschmid**
see commit message for explaination

### CODE REVIEW

File: client/src/components/Dashboard.tsx [hunk 1]
```diff
@@ -150,6 +150,7 @@ const Dashboard = () => {
                   rooms={rooms}
                   meetings={meetingsIndexed}
                   isDisabled={isDisabled}
+                  isJoinable={potentiallyDisabledGroup ? potentiallyDisabledGroup.isJoinable : true}
```
Comment (klassm) [inline client/src/components/Dashboard.tsx:153]: Can be shorted to
potentiallyDisabledGroup?.isJoinable ?? true

File: client/src/disabledGroups.ts [hunk 2]
```diff
@@ -4,15 +4,17 @@ import { DateTime } from "luxon";
...
   isUpcoming: boolean;
   isExpired: boolean;
+  isJoinable: boolean;
   group: Group;
 }
 
 function toPotentiallyDisabledGroup(group: Group): PotentiallyDisabledGroup {
   const now = DateTime.local();
   const isExpired = (group.disabledAfter && DateTime.fromISO(group.disabledAfter) <= now) || false;
+  const isJoinable = (group.joinableFrom) ? DateTime.fromISO(group.joinableFrom) <= now && !isExpired : !isExpired;
```
Comment (klassm) [inline client/src/disabledGroups.ts:14]: Not sure, but wouldn't this mean that upcoming cards are always enabled when joinableFrom is not set?

File: client/src/disabledGroups.spec.ts [hunk 3]
```diff
@@ -41,32 +47,41 @@ describe("disabledRooms", () => {
...
     const result = mapPotentiallyDisabledGroups(groups);
 
     expect(result).toEqual([
-      { group: disabledBefore, isUpcoming: true, isExpired: false },
-      { group: disabledAfter, isUpcoming: false, isExpired: true },
-      { group: disabledBeforeAndAfter, isUpcoming: true, isExpired: true },
-      { group: notDisabledBefore, isUpcoming: false, isExpired: false },
-      { group: notDisabledAfter, isUpcoming: false, isExpired: false },
-      { group: notDisabledBeforeAndAfter, isUpcoming: false, isExpired: false },
+      { group: disabledBefore, isUpcoming: true, isExpired: false, isJoinable: true },
```
Comment (klassm) [inline client/src/disabledGroups.spec.ts:76]: That's essentially the comment from below - do we really want that isJoinable is true here?

File: client/src/components/Dashboard.tsx (hunk 1 above)
Comment (aaschmid) [reply client/src/components/Dashboard.tsx:153]: But mine is more readable, isn't it?

File: client/src/disabledGroups.ts (hunk 2 above)
Comment (aaschmid) [reply client/src/disabledGroups.ts]: had fixed that already but the push was missing... should be correct now.

File: client/src/disabledGroups.spec.ts (hunk 3 above)
Comment (aaschmid) [reply client/src/disabledGroups.spec.ts]: see above, fixed with the additonal push

File: client/src/disabledGroups.spec.ts [hunk 4]
```diff
@@ -16,7 +16,12 @@ describe("disabledRooms", () => {
     clock = fakeTimers.install({ now: now });
   });
 
-  function groupFor(config: { disabledBefore?: number; disabledAfter?: number; name: string }): Group {
+  function groupFor(config: {
+    disabledBefore?: number;
+    disabledAfter?: number;
+    joinableFrom?: number;
```
Comment (klassm) [inline client/src/disabledGroups.spec.ts:22]: This should be probably joinableAfter, to be consistent with disabledAfter

Comment (aaschmid) [reply client/src/disabledGroups.spec.ts]: Ack, done

**Pull request merged**